File: Python/program/Dosen.py
# import class sivitas dari file sivitas
from Sivitas import Sivitas

# class declaration


class Dosen (Sivitas):
    # atribut private
    __nip = ""
    __fakultas = ""
    __pend_terakhir = ""
    __keahlian = ""

   # kons truktor
    def __init__(self, nik="", nama="", jenis_kelamin="", asal_univ="", email_edu="", nip="", fakultas="", pend_terakhir="", keahlian=""):
        super().__init__(nik, nama, jenis_kelamin, asal_univ, email_edu)
        self.__nip = nip
        self.__fakultas = fakultas
        self.__pend_terakhir = pend_terakhir
        self.__keahlian = keahlian

    # Getter and Setter

    def get_nip(self):
        return self.__nip

    def set_nip(self, nip):
        self.__nip = nip

    def get_fakultas(self):
        return self.__fakultas

    def set_fakultas(self, fakultas):
        self.__fakultas = fakultas

    def get_pend_terakhir(self):
        return self.__pend_terakhir

    def set_pend_terakhir(self, pend_terakhir):
        self.__pend_terakhir = pend_terakhir

    def get_keahlian(self):
        return self.__keahlian

    def set_keahlian(self, keahlian):
        self.__keahlian = keahlian

File: CPP/program/Dosen.cpp
// import file header
#include "header.hh"

/* konstruktor */

// konstruktor tanpa parameter
Dosen::Dosen()
{
    // set isi atribut secara default
    this->nip = "";
    this->fakultas = "";
    this->pend_terakhir = "";
    this->keahlian = "";
}
// konstruktor dengan parameter
Dosen::Dosen(string nik, string nama, string jenis_kelamin, string asal_univ, string email_edu, string nip, string fakultas, string pend_terakhir, string keahlian) : Sivitas(nik, nama, jenis_kelamin, asal_univ, email_edu)
{
    // set isi atribut berdasarkan isi parameter
    this->nip = nip;
    this->fakultas = fakultas;
    this->pend_terakhir = pend_terakhir;
    this->keahlian = keahlian;
}

/* Getter dan Setter*/

// Get nip
string Dosen::get_nip()
{
    return this->nip;
}

// Set nip
void Dosen::set_nip(string nip)
{
    this->nip = nip;
}

// Get fakultas
string Dosen::get_fakultas()
{
    return this->fakultas;
}

// Set fakultas
void Dosen::set_fakultas(string fakultas)
{
    this->fakultas = fakultas;
}
// Get pend_terakhir
string Dosen::get_pend_terakhir()
{
    return this->pend_terakhir;
}

// Set pend_terakhir
void Dosen::set_pend_terakhir(string pend_terakhir)
{
    this->pend_terakhir = pend_terakhir;
}
// Get keahlian
string Dosen::get_keahlian()
{
    return this->keahlian;
}

// Set keahlian
void Dosen::set_keahlian(string keahlian)
{
    this->keahlian = keahlian;
}

/* desktruktor */

// default
Dosen::~Dosen() // desktruktor
{
}
